src/main/java/org/elasticsearch/search/aggregations/reducers/derivative/DerivativeBuilder.java

```diff
@@ -45,6 +46,16 @@ public DerivativeBuilder gapPolicy(GapPolicy gapPolicy) {
         return this;
     }
 
+    public DerivativeBuilder units(String units) {
+        this.units = units;
+        return this;
+    }
+
+    public DerivativeBuilder units(DateHistogramInterval units) {
+        this.units = units.toString();
+        return this;
+    }
```

Comment:
The problem here is that this builder is for derivatives of either a `date_histogram` or a `histogram` so we need to expose a method of specifying units for the `histogram` as well as methods for the `date_histogram`